### DISCUSSION TITLE
How to revert changes related to HelmRealse

### DISCUSSION THREAD

**Discussion by rfalke** [fluxcd/flux2#3137] [Q&A]
We have a lot of Kustomization with for example Deployments. In such cases you can suspend flux for this Kustomization, test changes by editing the Deployment directly in k8s and do a resume afterwards to revert the changes.

For the few Kustomization which contain HelmReleases this approach doesn't work. You can edit the created resource and flux (even if nothing is suspended) doesn't revert the changes. The answer in https://stackoverflow.com/questions/65677606/is-there-a-way-to-manually-retry-a-helmrelease-for-fluxcd-helmoperator (resume, suspend) also doesn't revert the changes in the final resource. Our current approach is to delete (with k9s) objects like the HelmChart.

How can one trigger (with only the flux CLI) such a revert?

**Comment by kingdonb** [comment]
Currently the answer is, if you want Helm Controller to correct some drift in a Helm release that was manually introduced, then you need to trigger an "upgrade" operation somehow. You can do this by changing the inputs of the HelmRelease and reconciling it, generally, this should always work.

The `flux suspend helmrelease x; flux resume helmrelease x` trick should also work to trigger an upgrade, which would result in the drift being corrected. If that didn't work, that's surprising, but depending on what kind of drift also maybe not – certain things are created by hooks on "install" like secret generation and they would only be recreated on install.

We know that this situation around the drift not generally being automatically reconciled is not ideal and I hope we can report soon that Helm Controller also has drift correction like you'll find in the rest of Flux. The Helm release process is subtly imperative in a way that makes it prohibitive for Flux to truly do a continuous reconcile.

But I know there is a rewrite that's in progress that should correct this, please track this issue for more information:

* https://github.com/fluxcd/helm-controller/issues/186

Thanks for your patience.
